lib/tasks/scripts/create_master_user.rb

```diff
@@ -0,0 +1,77 @@
+module CreateMasterUser
+  def self.from_admin(admin)
+    user_id = master_user_id(admin.email)
+    return if User.find_by(id: user_id).present?
+
+    admin.transaction do
+      admin_attributes = admin.attributes.with_indifferent_access
+
+      master_user = create_user_for_admin!(admin)
+      email_authentication = EmailAuthentication.new(admin_attributes.except(:id, :role))
+      email_authentication.save!(validate: false)
+      master_user.user_authentications.create!(authenticatable: email_authentication)
+      assign_permissions!(master_user, admin)
+    end
+  end
+
+  private
+
+  def self.master_user_id(email)
+    UUIDTools::UUID.md5_create(
+      UUIDTools::UUID_DNS_NAMESPACE,
+      { email: email }.to_s
+    ).to_s
+  end
+
+  def self.create_user_for_admin!(admin)
+    admin_attributes = admin.attributes.with_indifferent_access
+
+    user_attributes =
+      admin_attributes
+        .slice(:role, :created_at, :updated_at, :deleted_at)
+        .merge(id: master_user_id(admin.email),
+               full_name: admin.email.split('@').first,
+               organization: get_organization(admin),
+               sync_approval_status: 'denied',
```

Comment:
Wouldn't it be nicer to have a 'n/a' status in the `sync_approval_status` enum? Right now, the status and the status reason seem a bit disjointed.